platform/api/controllers/SnippetsController.js

```diff
@@ -165,7 +167,18 @@ module.exports = {
         }
 
         return res.view({
+            mode: 'update',
             snippet
         });
+    },
+
+    async execute(req, res) {
+        let { language, source, stdin, argv } = req.body;
```

Comment:
This would either need to take the snippet id or we could just call the public API in snippet front-end. So it wouldn't be used to bypass the ratelimit.

Reply:
I would run it from the front-end, no need for additional endpoints